raiden/network/transport/matrix/transport.py

```diff
@@ -443,10 +451,12 @@ def stop(self):
             if retrier:
                 retrier.notify()
 
+        self._address_mgr.stop()
+        self._client.stop_listener_thread()  # stop sync_thread, wait on client's greenlets
```

Comment:
Shouldn't this happen after our state has become offline?

Reply:
No, unfortunately that doesn't work. You can't set yourself to offline while still communicating with the server. As soon as it sees another request it changes the state back to online.